src/machine/machine_atsamd51.go

```diff
@@ -1495,6 +1495,11 @@ func (spi SPI) tx(tx []byte) {
 	}
 	for !spi.Bus.INTFLAG.HasBits(sam.SERCOM_SPIM_INTFLAG_TXC) {
 	}
+
+	// read to clear RXC register
+	for spi.Bus.INTFLAG.HasBits(sam.SERCOM_SPIM_INTFLAG_RXC) {
+		spi.Bus.DATA.Get()
+	}
```

Comment:
This process is necessary because other functions are looking at RXC.

For example, if RXC is set to 1 at the time of the SPI.Transfer() call, the wait will fail.